llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtAttrs.cpp

```diff
@@ -217,9 +220,87 @@ bool LayoutAttr::hasLaneConflictWith(const LayoutAttr &other) {
   return false;
 }
 
+// Project the nested layout. This take a mask on the dimensions of the vector
+// associated with this layout and projects out those dimensions. This reduces
+// the rank of the layout in the process.
 VectorLayoutInterface
 NestedLayoutAttr::project(ArrayRef<bool> projectedDims) const {
-  llvm_unreachable("Not yet implemented");
+  assert(projectedDims.size() == getBatchesPerSubgroup().size() &&
+         "projectedDims size must match layout rank");
+
+  // Projection for this layout simply means the sizes along the projected
```

Comment:
I think projection typically means keep some dimensions _remaining_ rather than _dropping_? At that that way it's consistent with what projection means in MLIR (affine). So `projectedDims` here are `droppedDims` actually? Would be nice to rename if so.